pandas/core/algorithms.py

```diff
@@ -424,7 +425,7 @@ def nunique_ints(values: ArrayLike) -> int:
     return result
 
 
-def unique_with_mask(values, mask: npt.NDArray[np.bool_] | None = None):
+def unique_with_mask(values, mask: npt.NDArray[np.bool_] | BitmaskArray | None = None):
```

Comment:
havent looked that closely, but my kneejerk both here and in array_algos is that the caller should be responsible for converting the BitmaskArray (or do somethinh at a higher level that makes the conversion unnecessary)

Reply:
Yea I went back and forth with that design. What is nice about keeping this very local to the issue is that it is very clear where you would need to refactor in the future to make the BitmaskArray work better. In this case the only reason the conversion is needed is because of the `astype("bool")` call on L453. If that didn't exist we could probably avoid this isinstance check